src/Sylius/Bundle/ApiBundle/Controller/DeleteOrderItemAction.php

```diff
@@ -14,25 +14,34 @@
 namespace Sylius\Bundle\ApiBundle\Controller;
 
 use Sylius\Bundle\ApiBundle\Command\Cart\RemoveItemFromCart;
+use Sylius\Bundle\ApiBundle\Exception\OrderItemNotFoundException;
+use Sylius\Component\Core\Repository\OrderItemRepositoryInterface;
 use Symfony\Component\HttpFoundation\JsonResponse;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Messenger\MessageBusInterface;
 
 final class DeleteOrderItemAction
 {
-    public function __construct(private MessageBusInterface $commandBus)
+    public function __construct(
+        private MessageBusInterface $commandBus,
+        private OrderItemRepositoryInterface $orderItemRepository,
+    )
     {
     }
 
     public function __invoke(Request $request): Response
     {
-        $command = new RemoveItemFromCart(
-            $request->attributes->get('tokenValue'),
-            $request->attributes->get('itemId'),
-        );
+        $orderItemId = $request->attributes->get('itemId');
+        $tokenValue = $request->attributes->get('tokenValue');
 
-        $this->commandBus->dispatch($command);
+        $orderItem = $this->orderItemRepository->findOneByIdAndCartTokenValue($orderItemId, $tokenValue);
+
+        if ($orderItem === null) {
+            throw new OrderItemNotFoundException();
+        }
```

Comment:
I tried to put this logic inside the command handler, but the exception thrown there is caught by the middleware, which throws another exception that is not mapped to status 422 and request ends with 500. Throwing it here skips the middleware